src/Template/Layout/js/app/components/tag-picker/tag-picker.js

```diff
@@ -0,0 +1,114 @@
+import { Treeselect, ASYNC_SEARCH } from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+
+const API_OPTIONS = {
+    credentials: 'same-origin',
+    headers: {
+        'accept': 'application/json',
+    }
+};
+const QUERY_MIN_LENGTH = 4;
+
+export default {
+    components: {
+        Treeselect,
+    },
+
+    template: `
+        <div class="tag-picker">
+            <label v-if="label"><: label :></label>
+            <Treeselect
+                placeholder="Cerca tag esistenti"
+                :options="tagsOptions"
+                :async="true"
+                :clearable="false"
+                :load-options="fetchTags"
+                :disabled="disabled"
+                :disable-branch-nodes="true"
+                :multiple="true"
+                value-format="object"
+                v-model="selectedTags"
+                @select="onAdd"
+                @deselect="onRemove"
+            />
+            <div class="new-tag">
+                <label for="new-tag">Aggiungi un nuovo tag</label>
+                <div class="input-container">
+                    <input type="text" :value="text" id="new-tag" @input="update($event.target.value)" />
+                    <button @click.prevent="addNewTag">Aggiungi</button>
+                </div>
+            </div>
+            <input type="hidden" name="tags" :value="modifiedTags" />
+            <input type="hidden" name="_types[tags]" value="json" />
+        </div>
+    `,
+
+    props: {
+        disabled: Boolean,
+        label: String,
+        initialTags: Array,
+    },
+
+    data() {
+        return {
+            selectedTags: [],
+            modifiedTags: null,
+            tagsOptions: [],
+            text: '',
+        };
+    },
+
+    mounted() {
+        this.selectedTags = this.initialTags?.map((tag) => ({ id: tag.name, label: tag.label }));
+        this.parseBeforeSave();
+    },
+
+    methods: {
+        parseBeforeSave() {
+            this.modifiedTags = JSON.stringify(this.selectedTags.map((tag) => ({ name : tag.id, label: tag.label })));
+        },
```

Comment:
I think you can use a computed property instead of this method and the `modifiedTags` data prop. some other adjustment is required to trigger the lifecycle on `selectedTags` changes (replacing `splice` instead of `push` should work)